src/software/thunderscope/binary_context_managers.py

```diff
@@ -332,6 +332,17 @@ def __init__(self, supress_logs=False, ci_mode=False):
         self.supress_logs = supress_logs
         self.ci_mode = ci_mode
 
+    def __restart__(self):
```

Comment:
this needs to be called in a loop running on a separate thread

and then we'd start that thread after we have the first Popen call